libdrgn/python/program.c

```diff
@@ -1040,3 +1045,47 @@ Program *program_from_pid(PyObject *self, PyObject *args, PyObject *kwds)
 	}
 	return prog;
 }
+
+ThreadIterator *Program_threads(Program *self)
+{
+	Py_INCREF(self);
+	ThreadIterator *iterator =
+		(ThreadIterator *)ThreadIterator_type.tp_alloc(
+			&ThreadIterator_type, 0);
```

Comment:
When you say "this should return `NULL` to indicate an error", are you stating that `Program_threads` should return `NULL`, or that in the case that `tp_alloc` returns `NULL` we should return an error (likely `set_drgn_error(&drgn_enomem)`)?

Reply:
Python C API functions in general return NULL or -1 to indicate that an error has occurred and that they have set an exception. See https://docs.python.org/3/c-api/exceptions.html.

`tp_alloc` in particular returns `NULL` and sets an exception if it fails. If it does, this should also return `NULL`. You shouldn't set another exception, because you want to propagate the exception set by `tp_alloc`.